backend/tracim_backend/lib/utils/image_process.py

```diff
@@ -0,0 +1,57 @@
+from fractions import Fraction
+import typing
+
+from PIL import Image
+from PIL import ImageOps
+
+
+class ImageRatio:
+    def __init__(self, width: int, height: int):
+        self.width = width
+        self.height = height
+
+    def ratio(self) -> Fraction:
+        return Fraction(self.width, self.height)
+
+    def sized_to_ratio(self, image_size: "ImageSize") -> "ImageSize":
+        width = image_size.width
+        height = image_size.height
+        if image_size.ratio() == self.ratio():
+            pass
+        elif image_size.ratio() > self.ratio():
+            # INFO - G.M - 2021-01-21 - we do not allow new width to be
+            # bigger than previous one,
+            # in that specific edge case, we do not render proper ratio.
+            width = min(int(self.ratio() * height), width)
+        else:
+            # INFO - G.M - 2021-01-21 - we do not allow new height to be
+            # bigger than previous one,
+            # in that specific edge case, we do not render proper ratio.
+            height = min(int(Fraction(height, self.ratio())), height)
+
+        # INFO - G.M - 2021-01-21 - if we get 0 width or height,
+        # we should adjust to 1 to avoid error.
+        width = max(width, 1)
+        height = max(height, 1)
+        return ImageSize(width, height)
+
+
+class ImageSize(ImageRatio):
+    pass
```

Comment:
Why a separate class here? Maybe just name the previous class ImageDimension and use this everywhere?

Reply:
A ratio is not a size, so i do prefer a separate object.